docs/examples/azure-pipelines.md

```diff
@@ -0,0 +1,86 @@
+won't break due to updates to one of the addins.
+
+See [pinning addin versions](https://cakebuild.net/docs/tutorials/pinning-cake-version#pinning-addin-version) for details.
+:::
+
+In the following task we'll first determine if the build is running on Azure DevOps and for a pull request,
+then read the remote repository URL and pull request id from environment variables set by the Azure Pipelines build 
+and finally call the [TfsPullRequests] alias using the OAuth token provided by the Azure Pipeline build.
+
+:::{.alert .alert-info}
+Please note that you'll need to setup your Azure Pipelines build to allow scripts to
+access the OAuth token and need to setup proper permissions.
+
+See [OAuth authentication from Azure Pipelines] for details.
+:::
+
+```csharp
+Task("ReportIssuesToPullRequest").Does(() =>
+{
+    var isRunningOnAzureDevOps =
+        !string.IsNullOrWhiteSpace(context.EnvironmentVariable("TF_BUILD")) &&
+        !string.IsNullOrWhiteSpace(context.EnvironmentVariable("SYSTEM_COLLECTIONURI")) &&
+        (
+            new Uri(context.EnvironmentVariable("SYSTEM_COLLECTIONURI")).Host == "dev.azure.com" ||
+            new Uri(context.EnvironmentVariable("SYSTEM_COLLECTIONURI")).Host.EndsWith("visualstudio.com")
+        );
+
+    var isPullRequestBuild =
+        !string.IsNullOrWhiteSpace(context.EnvironmentVariable("SYSTEM_PULLREQUEST_PULLREQUESTID"));
+
+    if (isRunningOnAzureDevOps)
+        {
+            var repositoryUrl = new Uri(context.EnvironmentVariable("BUILD_REPOSITORY_URI"));
+
+            if (isPullRequestBuild)
+            {
+                if (!Int32.TryParse(context.EnvironmentVariable("SYSTEM_PULLREQUEST_PULLREQUESTID"), out var pullRequestId))
+                {
+                    throw new Exception(
+                        string.Format(
+                            "Invalid pull request ID: {0}",
+                            context.EnvironmentVariable("SYSTEM_PULLREQUEST_PULLREQUESTID")));
+                }
+                else
+                {
+                    var repoRootFolder = MakeAbsolute(Directory("./"));
+
+                    ReportIssuesToPullRequest(
+                        InspectCodeIssuesFromFilePath(
+                            @"C:\build\inspectcode.log"),
+                        TfsPullRequests(
+                            repositoryUrl,
+                            pullRequestId,
+                            TfsAuthenticationOAuth(EnvironmentVariable("SYSTEM_ACCESSTOKEN"))),
+                        repoRootFolder);
+                }
+            }
+        }
+});
+```
+
+[TfsPullRequests]: ../../../../api/Cake.Issues.PullRequests.Tfs/TfsPullRequestSystemAliases/BC3F9B2C
+[Allow scripts to access the OAuth token]: https://docs.microsoft.com/en-us/azure/devops/pipelines/build/options?view=vsts&tabs=yaml#allow-scripts-to-access-the-oauth-token
+[OAuth authentication from Azure Pipelines]: ../setup#oauth-authentication-from-azure-pipelines
```

Comment:
[MD047](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md047---files-should-end-with-a-single-newline-character) markdown files should end with a single newline character.